apps/browser/pages/index.tsx

```diff
@@ -192,13 +213,43 @@ function Dojo() {
     })();
   }, [dojoConfig]);
 
+  useEffect(() => {
+    if (!evo || dojoConfig.openAiApiKey) {
+      return
+    }
+
+    const proxyEmbedding = new EmbeddingProxy(cl100k_base, DEFAULT_ADA_CONFIG, goalId)
+    const proxyLlm = new LlmProxy(evo.context.env.GPT_MODEL, evo.context.env.CONTEXT_WINDOW_TOKENS, evo.context.env.MAX_RESPONSE_TOKENS, goalId)
+    evo.context.llm = proxyLlm
+    evo.context.embedding = proxyEmbedding
+    evo.recreateContextualizedChat(evo.context)
+  }, [goalId])
```

Comment:
this useEffect must be removed but since I am already touching a lot in this PR I decided to have this and then we can remove it, in a PR improving this entire file